### DISCUSSION TITLE
GUI/tkinter programming

### DISCUSSION THREAD

**Discussion by jorgemolongua** [exaloop/codon#181] [General]
Is GUI programming, like python's tkinter, supported?

**Comment by arshajii** [comment]
Hi @jorgemolongua -- yes, you can import e.g. `tkinter` as a Python library and call it from within Codon. For example:

``` python
from python import tkinter as tk

window = tk.Tk()
btn = tk.Button(window, text='a button', fg='blue')
btn.place(x=80, y=100)
window.title('hello world')
window.geometry('300x200+10+10')
window.mainloop()
```

Depending on the use case, it might make more sense to write the GUI code in plain Python and use Codon for the performance-sensitive bits via the [`@codon.jit` decorator](https://docs.exaloop.io/codon/interoperability/decorator).
